client/src/features/project/components/cloudStorage/AddOrEditCloudStorage.tsx

```diff
@@ -0,0 +1,1015 @@
+      setValue("mountPoint", `external_storage/${value}`);
+    setStorage({ ...getValues() });
+  };
+
+  return (
+    <form className="form-rk-green" data-cy="cloud-storage-edit-mount">
+      <h5>Final details</h5>
+      <p>We need a few more details to mount your storage properly.</p>
+
+      <div className="mb-3">
+        <Label className="form-label" for="add-storage-name">
+          Name
+        </Label>
+
+        <Controller
+          name="name"
+          control={control}
+          render={({ field }) => (
+            <input
+              id="name"
+              type="string"
+              {...field}
+              className={cx("form-control", errors.name && "is-invalid")}
+              onChange={(e) => {
+                field.onChange(e);
+                onFieldValueChange("name", e.target.value);
+              }}
+            />
+          )}
+          rules={{
+            // TODO: check this won't create a duplicate
+            validate: (value) =>
+              !value
+                ? "Please provide a name"
+                : /^[a-zA-Z0-9_-]+$/.test(value) ||
+                  "Name can only contain letters, numbers, underscores (_), and dashes (-)",
+          }}
+        />
+        <div className="invalid-feedback">
+          {errors.name?.message?.toString()}
+        </div>
+        <div className={cx("form-text", "text-muted")}>
+          This name will help you identify the storage. It should be unique for
+          this project and it can only contains letter, numbers, $, _.
+        </div>
+      </div>
+
+      <div className="mb-3">
+        <Label className="form-label" for="add-storage-name">
+          Mount point
+        </Label>
+
+        <Controller
+          name="mountPoint"
+          control={control}
+          render={({ field }) => (
+            <input
+              id="mountPoint"
+              type="string"
+              {...field}
+              className={cx("form-control", errors.mountPoint && "is-invalid")}
+              onChange={(e) => {
+                field.onChange(e);
+                onFieldValueChange("mountPoint", e.target.value);
+              }}
+            />
+          )}
+          rules={{ required: true }}
+        />
+        <div className="invalid-feedback">Please provide a mount point.</div>
+        <div className={cx("form-text", "text-muted")}>
+          This is the name of the folder where you will find your external
+          storage in the sessions. You should pick something different from the
+          folders used in the projects repository, and from folder mounted by
+          other storage services.
+        </div>
+      </div>
+
+      <div className="mb-3">
+        <Label className="form-label" for="add-storage-name">
+          Read-only
+        </Label>
+
+        <Controller
+          name="readOnly"
+          control={control}
+          render={({ field }) => (
+            <input
+              id="readOnly"
```

Comment:
The label is broken:
```suggestion
              id="add-storage-name"
```

Clicking the label should check/uncheck the option.